java/org/contikios/cooja/Simulation.java

```diff
@@ -152,68 +156,117 @@ public Simulation(Cooja cooja, long seed) {
     this.cooja = cooja;
     randomGenerator = new SafeRandom(this);
     randomSeed = seed;
+    simulationThread = new Thread(() -> {
+      boolean isAlive = true;
+      do {
+        boolean isSimulationRunning = false;
+        EventQueue.Pair nextEvent = null;
+        try {
+          while (isAlive) {
+            Object cmd;
+            do {
+              cmd = isSimulationRunning ? commandQueue.poll() : commandQueue.take();
+              if (cmd instanceof Runnable r) {
+                r.run();
+              } else if (cmd instanceof Command c) {
+                isAlive = c != Command.QUIT;
+                isShutdown = !isAlive;
+                isSimulationRunning = c == Command.START;
+                setRunning(isSimulationRunning);
+              }
+            } while (cmd != null && isAlive);
+
+            if (isSimulationRunning) {
+              /* Handle one simulation event, and update simulation time */
+              nextEvent = eventQueue.popFirst();
+              assert nextEvent != null : "Ran out of events in eventQueue";
+              assert nextEvent.time >= currentSimulationTime : "Event from the past";
+              currentSimulationTime = nextEvent.time;
+              nextEvent.event.execute(currentSimulationTime);
+            }
+          }
+        } catch (RuntimeException e) {
+          if ("MSPSim requested simulation stop".equals(e.getMessage())) {
+            /* XXX Should be*/
+            logger.info("Simulation stopped due to MSPSim breakpoint");
+          } else {
+            logger.fatal("Simulation stopped due to error: " + e.getMessage(), e);
+            if (!Cooja.isVisualized()) {
+              /* Quit simulator if in test mode */
+              System.exit(1);
+            } else {
+              String title = "Simulation error";
+              if (nextEvent != null && nextEvent.event instanceof MoteTimeEvent moteTimeEvent) {
+                title += ": " + moteTimeEvent.getMote();
+              }
+              Cooja.showErrorDialog(title, e, false);
+            }
+          }
+        } catch (InterruptedException e) {
+          // Simulation thread interrupted - quit
+          logger.warn("simulation thread interrupted");
+          isAlive = false;
+          isShutdown = true;
+        }
+        setRunning(false);
+      } while (isAlive);
+      isShutdown = true;
+      commandQueue.clear();
+
+      /* Deactivate all script engines */
```

Comment:
I think C99 comments stand out less, and the comments here for ending the simulation aren't something one should stop reading and pay special attention to.

Reply:
These comments are supported by C99. But I am guessing you mean `//` comments?
I used this type of comment here because the old comments in the block were of this type but I will change the comments I added.

Reply:
I think git thinks you added the comments on line 222 and 227 too.

Reply:
No, that is the original code (although moved from the `remove()` method) but I will modify it as requested.